File: Level 1/Scoreboard Inference 1/python/Scoreboard_Inference_1.py
from typing import List

def getMinProblemCount(N: int, S: List[int]) -> int:
    maxScore = -1
    containsOdd = False
    
    for score in S:
        if score > maxScore:
            maxScore = score
        containsOdd = containsOdd or (score % 2 == 1)
    
    return maxScore // 2 + (1 if containsOdd else 0)

File: Level 1/Scoreboard Inference 1/go/ScoreboardInference1.go
package main
import "bufio"
import "fmt"
import "os"
import "strconv"
import "strings"

func getMinProblemCount(N int32, S []int32) int32 {
  maxScore := int32(0);
  containsOdd := false;

	for i := int32(0); i < N; i++ {
		if S[i] > maxScore {
			maxScore = S[i];
		}

		containsOdd = containsOdd || S[i] % 2 == 1;
	}

	minProblemCount := maxScore / 2;
	if containsOdd {
		minProblemCount++;
	}

  return minProblemCount;
}
